### PR TITLE
ARROW-5865: [Release] Helper script to rebase PRs on master

### PR DESCRIPTION
**Pull request by emkornfield**
This make a script that will generate shell commands to force push all branchs.

### CODE REVIEW

**Comment by emkornfield** [comment]
CC @kou

**Comment by emkornfield** [comment]
Travis on my branch: https://travis-ci.org/emkornfield/arrow/builds/554767093?utm_medium=notification&utm_source=email

File: dev/release/generate_force_push_script.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from collections import defaultdict
+
+client = HTTPSConnection('api.github.com')
+client.request('GET',
+               '/repos/apache/arrow/pulls?state=open&per_page=100',
+               headers={'User-Agent': 'ApacheArrowRebaser'})
+response = client.getresponse()
+json_content = response.read()
+if response.status != 200:
+    error_msg = 'Github connection error:{}'.format(json_content)
```
Comment (kou) [inline dev/release/generate_force_push_script.py]: type: "Github" -> "GitHub"

File: dev/release/generate_force_push_script.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    raise Exception(error_msg)
+
+parsed_content = json.loads(json_content)
+if len(parsed_content) >= 100:
```
Comment (kou) [inline dev/release/generate_force_push_script.py]: `==` will be more readable.

File: dev/release/generate_force_push_script.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    raise NotImplementedError("The script only supports <= 100 PRs")
```
Comment (kou) [inline dev/release/generate_force_push_script.py]: How about just reporting a warning message to process at least the latest 100 PRs?

Anyway, we should not keep 100 over pull requests. :-)

File: dev/release/generate_force_push_script.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+for repo, labels in repos.items():
+    print('git clone git@github.com:{}.git'.format(repo))
+    print('cd arrow')
+    print('git remote add upstream https://github.com/apache/arrow.git')
+    print('git fetch --all --prune --tags --force')
+    for label in labels:
+        # Labels are in the form 'user:branch'
+        print('git checkout {}'.format(label[label.find(':')+1:]))
+        print('(git rebase upstream/master && git push --force) || ' +
+              '(echo "Rebase failed for {} && '.format(label) +
```
Comment (kou) [inline dev/release/generate_force_push_script.py]: Closed `"` is missing for `echo`.

File: dev/release/generate_force_push_script.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+              'git rebase --abort)')
+        print('git push --force')
```
Comment (kou) [inline dev/release/generate_force_push_script.py]: This `push` may be redundant.

File: dev/release/generate_force_push_script.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    repos[head['repo']['full_name']].append(head['label'])
+
+for repo, labels in repos.items():
+    print('git clone git@github.com:{}.git'.format(repo))
+    print('cd arrow')
+    print('git remote add upstream https://github.com/apache/arrow.git')
+    print('git fetch --all --prune --tags --force')
+    for label in labels:
+        # Labels are in the form 'user:branch'
+        print('git checkout {}'.format(label[label.find(':')+1:]))
```
Comment (kou) [inline dev/release/generate_force_push_script.py]: Assigning to variables may be more readable than left a comment:

```python
user, branch = label.split(':')
print('git checkout {}'.format(branch))
```

Comment (kou) [reply dev/release/generate_force_push_script.py]: I don't try this yet but we may need to use `git checkout -b branch origin/branch`.

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
This script will help us in the future releases!

File: dev/release/generate_force_push_script.py (hunk 1 above)
Comment (emkornfield) [reply dev/release/generate_force_push_script.py]: Done.

File: dev/release/generate_force_push_script.py (hunk 2 above)
Comment (emkornfield) [reply dev/release/generate_force_push_script.py]: Done.

File: dev/release/generate_force_push_script.py (hunk 3 above)
Comment (emkornfield) [reply dev/release/generate_force_push_script.py]: done.  of course we will always keep the PR queue small :)

File: dev/release/generate_force_push_script.py (hunk 5 above)
Comment (emkornfield) [reply dev/release/generate_force_push_script.py]: it is.  removed.

File: dev/release/generate_force_push_script.py (hunk 4 above)
Comment (emkornfield) [reply dev/release/generate_force_push_script.py]: done.

File: dev/release/generate_force_push_script.py (hunk 6 above)
Comment (emkornfield) [reply dev/release/generate_force_push_script.py]: done for the first.  The current formulation worked for me, but I don't know if it is git version dependent.

**Review by kou — APPROVED** [review]
+1
We can try this in the next release and create `dev/release/post-XXX.sh` to automate this task.

**Pull request closed without merging**